dbt/include/global_project/macros/adapters/snowflake.sql

```diff
@@ -1,6 +1,6 @@
 {% macro snowflake__create_table_as(temporary, relation, sql) -%}
   {% if temporary %}
-    use schema {{ schema }};
+    use schema {{ adapter.quote(schema) }};
```

Comment:
I don't think this is exactly right -- we don't _always_ want to quote this schema name. Instead, we want to use the quoting policy defined for the project (and therefore, the model).

The difference can arise if you have schema quoting turned off with a lowercase schema name on Snowflake. The actual schema would be `schema_name`, which Snowflake interprets as `SCHEMA_NAME`. This statement would be `use schema "schema_name"`, which would not exist!

It's a little difficult to reason about the nuances cross-db, so I might have the wrong idea here. Happy to discuss!